File: Java/com.faculdadevincit.designpatterns.strategypattern/src/PagamentoPix.java
public class PagamentoPix implements IPagamentoStrategy {

    @Override
    public void pagar(double valor) {
        double desconto = valor * .10;
        double valorComDesconto = valor - desconto;
        System.out.println("Pagamento de R$ " + valorComDesconto + " realizado com sucesso!");
        System.out.println("Desconto por pagamento à vista: R$ " + desconto);
    }

}

File: CSharp/FaculdadeVincit.DesignPatterns.StrategyPattern/PagamentoPix.cs
namespace FaculdadeVincit.DesignPatterns.StrategyPattern;

public class PagamentoPix : IPagamentoStrategy
{
    public void Pagar(double valor)
    {
        var desconto = valor * .10;
        var valorComDesconto = valor - desconto;
        Console.WriteLine($"Pagamewnto de R$ {valorComDesconto} realizado com Pix!");
        Console.WriteLine($"Desconto por pgto. à vista: R$ {desconto}");
    }

}
